zetaclient/zetacore_observer.go

```diff
@@ -317,15 +326,48 @@ func (co *CoreObserver) scheduleCctxBTC(
 	}
 }
 
-func (co *CoreObserver) getUpdatedChainOb(appContext *appcontext.AppContext, chainID int64) (interfaces.ChainClient, error) {
+// GetUpdatedSigner returns signer with updated chain parameters
+func (co *CoreObserver) GetUpdatedSigner(coreContext *corecontext.ZetaCoreContext, chainID int64) (interfaces.ChainSigner, error) {
+	signer, found := co.signerMap[chainID]
+	if !found {
+		return nil, fmt.Errorf("signer not found for chainID %d", chainID)
+	}
+	// update EVM signer parameters only
+	if common.IsEVMChain(chainID) {
+		evmSigner, ok := signer.(*evm.Signer)
+		if !ok {
+			return signer, fmt.Errorf("signer is not an EVM signer for chainID %d", chainID)
+		}
+		evmParams, found := coreContext.GetEVMChainParams(chainID)
+		if found {
+			// update zeta connector and ERC20 custody addresses
+			zetaConnectorAddress := ethcommon.HexToAddress(evmParams.GetConnectorContractAddress())
+			erc20CustodyAddress := ethcommon.HexToAddress(evmParams.GetErc20CustodyContractAddress())
+			if zetaConnectorAddress != evmSigner.GetZetaConnectorAddress() {
+				evmSigner.SetZetaConnectorAddress(zetaConnectorAddress)
+				co.logger.ZetaChainWatcher.Info().Msgf(
+					"updated zeta connector address for chainID %d, new address: %s", chainID, zetaConnectorAddress)
+			}
+			if erc20CustodyAddress != evmSigner.GetERC20CustodyAddress() {
+				evmSigner.SetERC20CustodyAddress(erc20CustodyAddress)
+				co.logger.ZetaChainWatcher.Info().Msgf(
+					"updated ERC20 custody address for chainID %d, new address: %s", chainID, erc20CustodyAddress)
+			}
+		}
+	}
+	return signer, nil
+}
+
+// GetUpdatedChainOb returns chain client with updated chain parameters
+func (co *CoreObserver) GetUpdatedChainOb(coreContext *corecontext.ZetaCoreContext, chainID int64) (interfaces.ChainClient, error) {
```

Comment:
Can we add unit test for this method?

Reply:
https://github.com/zeta-chain/node/pull/1890/commits/1321dccf1e8bc841a1d84e04ae3ea9140ad2ed74